### PR TITLE
Cy 416 cyences severity

### PR DESCRIPTION
**Pull request by mahirchavda**
Add cyences_severity field for all the alerts

### CODE REVIEW

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 1]
```diff
@@ -917,12 +948,14 @@ request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = `ms_obj_changes_base_cat_customised("userchanged")`\
 | `ms_obj_user_change_out`\
+| eval cyences_severity = case( in(dest_user_subject, `cs_ad_important_user`), "medium", true(), "low")  \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:951]: Can we add some automation to figure out user is a high priority for the user or not?
If we can not get it directly, let's create a Jira ticket for the future.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 2]
```diff
@@ -950,12 +983,14 @@ search = `cs_wineventlog_security` (EventCode=4740) \
 | fields ComputerName Account_Domain Account_Name Caller_Computer_Name \
 | rename ComputerName as "Domain_Controller", Account_Domain as User_Domain ,Account_Name as "User_Name", Caller_Computer_Name as "Source_Computer_Name" \
 | fillnull value="-" Source_Computer_Name \
-| table _time,User_Name,User_Domain,Source_Computer_Name,Domain_Controller \
+| eval cyences_severity = "medium" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:986]: If we cannot determine whether user is high priority user or not in automated fashion let's create Jira ticket for the future.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -1912,6 +2006,7 @@ action.cyences_notable_event_action.param.filter_macro_name = cs_o365_failed_log
 action.cyences_notable_event_action.contributing_events = `cs_o365` sourcetype="o365:management:activity"  Workload=AzureActiveDirectory Operation=UserLoginFailed LogonError="BlockedByConditionalAccess"
 action.cyences_notable_event_action.system_compromised_search = | stats count by user
 action.cyences_notable_event_action.system_compromised_drilldown = `cs_o365` sourcetype="o365:management:activity"  Workload=AzureActiveDirectory Operation=UserLoginFailed LogonError="BlockedByConditionalAccess" user=$row.user$
+action.cyences_send_email_action = 1
 
 [O365 - Daily Login Failure]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2011]: This is a kind of daily report. I think it should be included on the overview page. I'm not sure 100%

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 4]
```diff
@@ -1607,10 +1679,12 @@ search = `cs_o365_managementactivity_change("Group")` _index_earliest=-61m@m _in
 | `cs_o365_managementactivity_change_out` \
 | `cs_o365_group_change_internal_filter` \
 | `cs_o365_group_change_filter`  \
-| table event_time, Actor, Command, Target_ObjectID, Target_DisplayName, ResultStatus, ModifiedProperties, MetaData
+| eval cyences_severity = case( in(Target_DisplayName, `cs_ad_important_group`), "medium", true(), "low")  \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1682]: Let's add any default groups like Global Admin, etc in here by default.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 5]
```diff
@@ -1637,10 +1711,12 @@ search = `cs_o365_managementactivity_change("GroupMembership")` _index_earliest=
 | `cs_o365_managementactivity_change_out` \
 | `cs_o365_group_membership_change_internal_filter` \
 | `cs_o365_group_membership_change_filter`  \
-| table event_time, Actor, Command, Target_ObjectID, Target_DisplayName, ResultStatus, ModifiedProperties, MetaData
+| eval cyences_severity = case( in(Target_DisplayName, `cs_ad_important_group`), "medium", true(), "low")  \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1714]: Let's add any default groups like Global Admin, etc in here by default.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 6]
```diff
@@ -1667,10 +1743,12 @@ search = `cs_o365_managementactivity_change("User")` _index_earliest=-61m@m _ind
 | `cs_o365_managementactivity_change_out` \
 | `cs_o365_user_change_internal_filter` \
 | `cs_o365_user_change_filter`  \
-| table event_time, Actor, Command, Target_ObjectID, Target_DisplayName, ResultStatus, ModifiedProperties, MetaData
+| eval cyences_severity = case( in(Target_DisplayName, `cs_ad_important_user`), "medium", true(), "low")  \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1746]: Let's add any default users like Global System User, etc in here by default.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 7]
```diff
@@ -2491,7 +2617,8 @@ display.page.search.mode = fast
 request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = `cs_crowdstrike_eventstream` "metadata.eventType"=DetectionSummaryEvent action="allowed" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2619]: Let's keep both allowed and not allowed.
But for allowed keep the severity as it is. For not-allowed keep the severity as info.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 8]
```diff
@@ -178,6 +180,7 @@ search = | tstats `cs_summariesonly_endpoint` count min(_time) as firstTime max(
 |  `cs_system_processes_run_from_unexpected_locations_internal_filter` \
 | lookup cs_fake_windows_processes_filter process_name, process_hash OUTPUT need_to_filtered | search NOT need_to_filtered=1 \
 | `is_windows_system_file` \
+| eval cyences_severity = "low" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:183]: Rename to "Windows - Fake Windows Processes"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 9]
```diff
@@ -213,6 +217,7 @@ request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = | tstats `cs_summariesonly_network_traffic` count min(_time) as firstTime max(_time) as lastTime from datamodel=Network_Traffic where All_Traffic.app=tor AND All_Traffic.action=allowed by All_Traffic.src_ip All_Traffic.dest_ip All_Traffic.dest_port All_Traffic.action \
 | `cs_human_readable_time_format(firstTime)` | `cs_human_readable_time_format(lastTime)` | `drop_dm_object_name("All_Traffic")` \
+| eval cyences_severity = "critical" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:220]: Network - TOR Traffic

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 10]
```diff
@@ -355,6 +367,7 @@ request.ui_dispatch_view = search
 search = | tstats `cs_summariesonly_endpoint` count values(Processes.process) as process values(Processes.parent_process) as parent_process min(_time) as firstTime max(_time) as lastTime from datamodel=Endpoint.Processes where Processes.process_name=fsutil.exe by Processes.user Processes.process_name Processes.parent_process_name Processes.dest \
 | `drop_dm_object_name(Processes)` | `cs_human_readable_time_format(firstTime)` | `cs_human_readable_time_format(lastTime)` \
 | search process="*deletejournal*" AND process="*usn*" \
+| eval cyences_severity = "critical" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:370]: Rename alert to "Windows - USN Journal Deletion on Windows"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 11]
```diff
@@ -363,6 +376,7 @@ action.cyences_notable_event_action.system_compromised_search = | stats sum(coun
 action.cyences_notable_event_action.system_compromised_drilldown = | datamodel Endpoint Processes search strict_fields=false | search Processes.dest=$row.dest$ | `drop_dm_object_name(Processes)`
 action.cyences_notable_event_action.attacker_search = | stats sum(count) as count, values(process) as process by process_name, parent_process_name
 action.cyences_notable_event_action.attacker_drilldown = | datamodel Endpoint Processes search strict_fields=false | search Processes.process_name=$row.process_name$ | `drop_dm_object_name(Processes)`
+action.cyences_send_email_action = 1
 
 
 [Ransomware - Windows - Windows Event Log Cleared]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:382]: Rename to "Windows - Windows Event Log Cleared"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 12]
```diff
@@ -413,6 +428,7 @@ action.cyences_notable_event_action.system_compromised_search = | stats sum(coun
 action.cyences_notable_event_action.system_compromised_drilldown = ((`cs_wineventlog_security` (EventCode=1102 OR EventCode=1100)) OR (`cs_wineventlog_system` EventCode=104)) dest=$row.dest$
 action.cyences_notable_event_action.attacker_search = | stats sum(count) as count by user
 action.cyences_notable_event_action.attacker_drilldown = ((`cs_wineventlog_security`) OR (`cs_wineventlog_system`)) user=$row.user$
+action.cyences_send_email_action = 1
 
 
 [Ransomware - Endpoint Compromise - Windows - WMI Lateral Movement]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:434]: Rename "Windows - WMI Lateral Movement"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 13]
```diff
@@ -722,12 +740,14 @@ request.ui_dispatch_app = cyences_app_for_splunk
...
 | `cs_windows_host_missing_update_filter`
 action.cyences_notable_event_action = 1
 action.cyences_notable_event_action.param.filter_macro_name = cs_windows_host_missing_update_filter
 action.cyences_notable_event_action.system_compromised_search = | dedup host | table host
 action.cyences_notable_event_action.system_compromised_drilldown = `cs_wineventlog_system` EventCode=19 SourceName="Microsoft-Windows-WindowsUpdateClient" host=$row.host$
-
+action.cyences_send_email_action = 1
+action.cyences_send_email_action.param.cyences_severities = critical, high
 
 [Windows - Endpoint Compromise - Windows Firewall Disabled Event]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:752]: Rename "Windows - Windows Firewall Disabled Event"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 14]
```diff
@@ -1079,6 +1120,7 @@ request.ui_dispatch_view = search
 search = index=* ( (tag=network tag=communicate) OR sourcetype=pan*traffic OR sourcetype=opsec OR sourcetype=cisco:asa) \
 | stats dc(dest_port) as num_dest_port dc(dest_ip) as num_dest_ip by sourcetype, src_ip\
 | where num_dest_port > 100 OR num_dest_ip > 500\
+| eval cyences_severity = case(num_dest_ip>1000 and num_dest_port>1000, "critical", num_dest_ip>1000 or num_dest_port>1000, "high", true(), "medium") \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1123]: Rename alert to "Network - Basic Scanning"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 15]
```diff
@@ -1116,6 +1159,7 @@ request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = `cs_palo` (sourcetype="pan:traffic" OR sourcetype="pan:threat") dest_ip="72.5.65.111" \
 | stats count, values(dvc) as dvc, values(dvc_name) as dvc_name, dc(src_port) as dc_src_port, values(dest_port) as dest_port, values(url) as url, values(app) as app, values(rule) as rule, values(http_category) as http_category by src_ip \
+| eval cyences_severity = "medium" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1162]: Rename "Palo Alto Firewall - Palo Alto DNS Sinkhole"

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 16]
```diff
@@ -1147,7 +1192,13 @@ display.page.search.mode = fast
 request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = `cs_palo` sourcetype="pan:threat" severity IN ("high" "critical") \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1194]: Let's remove the severity filter.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 17]
```diff
@@ -1180,14 +1232,16 @@ display.page.search.mode = fast
 request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = `cs_palo` sourcetype="pan:system" severity IN ("critical", "high") NOT "license for" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1234]: Let's remove the severity filter.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 18]
```diff
@@ -2016,10 +2115,12 @@ search = `cs_o365` sourcetype="ms:o365:reporting:messagetrace" _index_earliest=-
     | eval average_emails_per_7days=round(average_emails_per_7days,2) ] \
 | eval PercentIncrease = (count - average_emails_per_7days) / count * 100 \
 | where count > average_emails_per_7days AND PercentIncrease > `cs_email_increase_over_baseline_limit_in_percentage` AND count > 10 \
+| eval cyences_severity = case(count>10*average_emails_per_7days, "critical", count>5*average_emails_per_7days, "high", count>2*average_emails_per_7days, "medium", true(), "low") \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2118]: This alert is specific to Office 365 but the alert name suggests a generic naming convention.
Let's discuss this.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 19]
```diff
@@ -2043,6 +2144,7 @@ request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
 search = `cs_o365` sourcetype="ms:o365:reporting:messagetrace" _index_earliest=-25h@m _index_latest=-1h@m Status=FilteredAsSpam \
 | stats count, values(RecipientAddress) as RecipientAddress by SenderAddress, Subject \
+| eval cyences_severity = "info" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2147]: This alert is specific to Office 365 but the alert name suggests a generic naming convention.
Let's discuss this.

**Review by VatsalJagani — CHANGES_REQUESTED** [review]
Hold off regarding alert rename-related comments.
- Let's add some category-related parameters as alert_action param instead.
- And simplify the alert names.

**Comment by mahirchavda** [comment]
> Hold off regarding alert rename-related comments.
> 
> * Let's add some category-related parameters as alert_action param instead.
> * And simplify the alert names.

Right. I would also suggest not to combine it with this PR. I will create a jira you can mention update Jira with alert name changes. Jira: CY-425

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 1 above)
Comment (mahirchavda) [reply cyences_app_for_splunk/default/savedsearches.conf:951]: Currently, there is no user_type field for the AD events. Also, checked the cs_ad_obj_user lookup but all user has the user_type=user value so need to find a way to achieve this. Created Jira: CY-426

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 3 above)
Comment (mahirchavda) [reply cyences_app_for_splunk/default/savedsearches.conf:2011]: It's already included in the overview page.

**Review by VatsalJagani — APPROVED** [review]

**Pull request merged**